security/mobile-app-network-security.md

```diff
@@ -0,0 +1,126 @@
+
+# Use Cases
+
+The main use cases that will be discussed in this document are:
+
+1. As a mobile developer, I want to use AeroGear services in my mobile apps. I have the AeroGear service running locally with HTTPS, and I want to test against the local services from my mobile app on an actual device.
+
+2. As an enterprise mobile developer, I need to develop a mobile app for internal use. I need to make sure my mobile app can connect to AeroGear services that are running with the organisation's own HTTPS certificates.
+
+3. As a mobile developer of a finance institution, I need to develop a mobile app that uses the AeroGear services for the customers of the organisation. However, due to highly sensitive nature of the data, I need to make sure that no one (including the app users themselves) can inspect the network data of the app.
+
+# Problem Description
+
+The above use cases can be categoried into 2 problems:
+
+1. Use case 1 & 2 are about allowing mobile apps consume AeroGear services that use self-signed certificates
+
+2. Use case 3 is about preventing Man-in-the-middle attack when mobile apps communicate with AeroGear services
+
+## Problems Explained
+
+If you already know quite well about the problems, then you can skip this section. 
+
+### Problem 1: Support self-signed certificates
+
+To consume AeroGear services, mobile apps will use the AeroGear SDKs, which use HTTP/HTTPS to communicate with the backend services. In most cases, HTTPS is used to ensure the security of the data during transimission. 
+
+When the HTTPS connection is established, the HTTP client on the device will need some way to verify the the authencitiy of the server. This is done by checking the server's certificate during the handshake between the client and the server. 
+
+In most cases, the server should have a certificate that is signed by a trusted CA to confirm the identity of the server. The HTTP client can then use the trusted CAs' certificates that are pre-installed on the device to verify the server's certificate.
+
+However, due to various reasons (like cost, and sometimes it is just not necessary), CA-signed certificates are not always being used for HTTPS connections. Instead, self-signed certificates are being used. But in this case, the HTTP client on the mobile device will not be able to verify the authentity of the server, and it will refuse to connect to the server. 
+
+In use case #1, when the AeroGear services are running locally, they will be using self-signed certificates by default. This means developers will not be able to connect to these services from their mobile devices by default.
+
+To solve this problem, all the modern OSes allow users install self-signed CA certificates on their devices and trust them. However, there is a side effect: it is possible for attackers to take advantages of it and perform man-in-the-middle attacks.
```

Comment:
'all the modern OSes' > 'Android & ioS' ?

Reply:
should be `all the modern mobile OSes` and yes there are more than `Android and iOS`

Reply:
But do we intend to, as part of doing this proposal, solve the problem for anything other than Android and iOS?

Reply:
no strong opinion given it's not about our solution :+1: